### DISCUSSION TITLE
Why not used with SwiftUI View

### DISCUSSION THREAD

**Discussion by Foixa** [hmlongco/Factory#66] [Q&A] [answered]
I've reviewed a lot of open source projects using Factory. None of them register a SwiftUI view. Is there any reason for this?Generally, repositories and services are registered. 

Do you have a personal recommendation regarding this?

For example, is there a problem in using it like this?

```
var homeView: Factory<some View> {
  shared { HomeView() }
}
```

**Comment by hmlongco** [comment, marked as answer]
Factory is used in SwiftUI to provide services to views and view models. I wouldn't recommend creating views with it, especially considering how often SwiftUI rebuilds and diff's the view tree in response to state changes.
